### PR TITLE
[1] Add request factory

### PR DESCRIPTION
**Pull request by e7mac**
PLAN

Done in this PR
1. Create Protocols / Interfaces for the class that we want to refactor. eg. Create LPRequesting 
2. Set the old class to conform to these new Protocols. LeanplumRequest conforms to LPRequesting

Future
3. All uses of LeanplumRequest become uses of objects that conform to LPRequesting
4. Switch the creation of LeanplumRequest objects through a new LPRequestFactory
5. Create new class LPRequest that conforms to LPRequesting
6. Use a feature flag to vend LPRequest instead of LeanplumRequest
7. Control the rollout using the feature flag

### CODE REVIEW

File: Leanplum-SDK/Classes/Managers/Networking/LPRequestFactory.m [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+@implementation LPRequestFactory
+
++ (id<LPRequesting>)get:(NSString *)apiMethod params:(NSDictionary *)params {
+//    LPLogType level = [apiMethod isEqualToString:LP_METHOD_LOG] ? LPDebug : LPVerbose;
+//    LPLog(level, @"Will call API method %@ with arguments %@", apiMethod, params);
+    return [[LeanplumRequest alloc] initWithHttpMethod:@"GET" apiMethod:apiMethod params:params];
+}
+
++ (id<LPRequesting>)post:(NSString *)apiMethod params:(NSDictionary *)params {
```
Comment (MaitreyaBuddha) [inline Leanplum-SDK/Classes/Managers/Networking/LPRequestFactory.m]: This API is a little strange. `LPRequestFactory.get` intuitively sounds like it's getting a request. Not creating a `GET` request. Maybe something like `makeGet` and `makePost` https://en.wikipedia.org/wiki/Factory_method_pattern

File: Leanplum-SDK/Classes/Managers/Networking/LPRequestFactory.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#import <Foundation/Foundation.h>
+#import "LPRequesting.h"
+
+@interface LPRequestFactory : NSObject
+
++ (id<LPRequesting>)get:(NSString *)apiMethod params:(NSDictionary *)params;
```
Comment (MaitreyaBuddha) [inline Leanplum-SDK/Classes/Managers/Networking/LPRequestFactory.h]: I'm not sure why we are using the factory pattern here if we're really just defining static `createGetRequst` and `createPostRequest` methods.

**Comment by e7mac** [comment]
test this please

**Comment by e7mac** [comment]
test this please

**Pull request merged**